docs/DeveloperGuide.md

```diff
@@ -99,28 +109,92 @@ Optionally, you can follow the [Using Checkstyle](UsingCheckstyle.md) document t
+
+`window.app` is broken down into two main parts
+- the summary view
+- and the tabbed interface
+
+Summary view act as the main dashboard which shows the various calculations. </br>
+Tabbed interface is responsible for loading various modules such as authorship to display additional information.
+
+### Javascript Files
+- [**main.js**](../frontend/src/static/js/main.js) - main controller that pushes content into different modules
+- [**api.js**](../frontend/src/static/js/api.js)- loading and parsing of the dashboard content
+- [**v_summary.js**](../frontend/src/static/js/main.js) - module that supports the ramp chart view
+- [**v_authorship.js**](../frontend/src/static/js/main.js) - module that supports the authorship view
+
+### Main (main.js)
+This contains the logic for main VueJS object, `window.app`, which is responsible for passing the necessary data into the relevant modules to be loaded.
+
+#### Loading of dashboard information
+The main Vue object depends on the `summary.json` data to determine the right `commits.json` files to load into memory. This is handled by `api.js` which loads the relevant file information from the network files if it is available, otherwise a report archive, `archive.zip`, have to be used.
+
+Once the relevant `commit.json` files are loaded, all the repo information will be passed into `v_summary` to be loaded in the summary view as the relevant ramp charts.
+
+#### Activating additional view modules
+Most activity or actions should happen within the module itself, but in the case where there is a need to control the tab view from the module, an event is emitted from the module to the main Vue object (`window.app`), which then handles the data received and passes it along to the relevant modules.
+
+An additional thing to note is the event handler should handle the switching of the tabs as well, activating the display of the rights tabs just that the relevant tabs are being displayed at all times.
+
+#### Hash link
+Other than the global main Vue object, another global variable we have is the `window.hashParams`. This object is reponsible for generating the relevant permalink for a specific view of the summary module for the dashboard.
+
+### Data loader (api.js)
+This is the module that is in charged of loading and parsing the data files generated as part of the report.
+
+#### Loading from ZIP file
+Due security design, most modern browsers do not allow web pages to obtain local files using the directory alone. As such, a ZIP archive of the report information will be produced alongside the report generation.
+
+This archive can be used in place of the network files to load the information into the dashboard when the dashboard HTML.
+
+As such, the API module will be handling all request for all the JSON data files. If the network file is not available, the files will be obtained from the zip archive provided.
```

Comment:
Capital **ZIP**
No need for `As such` in the beginning as well, this sentence doesn't link to the previous right?